### PR TITLE
[EP-674] Added iDEAL cells and view model

### PR DESCRIPTION
**Pull request by mpetrenco**
### Note:
Please don't look too hard at `JPSectionView`. It has been rewritten and replaced with a better version.

---

### CODE REVIEW

File: Source/Modules/PaymentMethods/View/Subviews/Cells/iDEALBankCell/JPPaymentMethodsIDEALBankCell.m [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+const float kiDEALBankVerticalPadding = 18.0f;
+const float kiDEALBankStackViewHorizontalPadding = 27.0f;
+const float kiDEALBankStackViewHeight = 24.0f;
+const float kiDEALBankIconWidth = 60.0f;
+const float kiDEALBankCheckmarkWidth = 34.0f;
+const float kiDEALBankSeparatorHeight = 1.0f;
+const float kiDEALBankStackViewSpacing = 10.0f;
+
+
```
Comment (stpopa) [inline Source/Modules/PaymentMethods/View/Subviews/Cells/iDEALBankCell/JPPaymentMethodsIDEALBankCell.m]: extraline

File: Source/View/SectionView/JPSectionView.m [hunk 2]
```diff
@@ -87,6 +90,16 @@ - (void)removeSections {
     }
 }
 
+- (void)changeSelectedSectionToIndex:(NSUInteger)index {
+    UIStackView *selectedStackView = self.mainStackView.arrangedSubviews[index];
+    UILabel *selectedLabel = selectedStackView.subviews[1];
```
Comment (stpopa) [inline Source/View/SectionView/JPSectionView.m]: .firstObject ?

File: Source/View/SectionView/JPSectionView.m [hunk 3]
```diff
@@ -400,7 +413,7 @@ - (UIStackView *)mainStackView {
 - (UIView *)selectorView {
     if (!_selectorView) {
         _selectorView = [UIView new];
-        _selectorView.frame = CGRectMake(0, 0, 100, sectionHeight);
+        _selectorView.frame = CGRectMake(0, 0, 100, kSectionViewSectionHeight);
```
Comment (stpopa) [inline Source/View/SectionView/JPSectionView.m]: lets extract please the corner radius and width to constants

File: Source/View/SectionView/JPSectionView.m [hunk 4]
```diff
@@ -308,21 +321,21 @@ - (void)registerGestureRecognizers {
...
+        backgroundViewWidth = screenWidth - (kSectionViewHorizontalPadding * 4);
         [self appendLongTextWidthForContentWidth:backgroundViewWidth];
     }
     return backgroundViewWidth;
 }
 
 - (CGFloat)calculateMainStackViewWidth {
     CGFloat screenWidth = UIScreen.mainScreen.bounds.size.width;
-    CGFloat mainStackViewWidth = screenWidth - (horizontalPadding * 2) - 50;
+    CGFloat mainStackViewWidth = screenWidth - (kSectionViewHorizontalPadding * 2) - 50;
```
Comment (stpopa) [inline Source/View/SectionView/JPSectionView.m]: extract please `50` to a constant

File: Source/View/SectionView/JPSectionView.m (hunk 2 above)
Comment (mpetrenco) [reply Source/View/SectionView/JPSectionView.m]: arrays start at 0 :(

**Pull request merged**